components/goals/GoalItem.js

```diff
@@ -0,0 +1,64 @@
+import Card from '../ui/Card';
+import GoalDetailModal from './GoalDetailModal';
+import PropTypes from 'prop-types';
+import { Typography, Button, Divider } from '@mui/material';
+import { useState } from 'react';
+
+function GoalItem({ title, owner, isEmployer }) {
+  const [open, setOpen] = useState(false);
+
+  function handleDetails() {
+    setOpen(true);
+  }
+
+  //TODO replace with an endpoint to approve the goal
+  function handleApprove() {
+    console.log('Approved');
+  }
+
+  //TODO replace with an endpoint to reject the goal
+  function handleReject() {
+    console.log('Rejected');
+  }
+
+  //TODO modify to enable editing
+  function handleEdit() {
+    console.log('to edit');
+  }
+
+  //TODO replace with an endpoint to delete the goal
+  function handleDelete() {
+    console.log('deleted');
+  }
+
+  function handleClose() {
+    setOpen(false);
+  }
+
+  return (
+    <Divider>
+      <Card>
+        <Divider>
+          <Typography variant="h3">{title}</Typography>
+          <Typography variant="h2">{owner}</Typography>
+        </Divider>
+        <Divider>
+          <Button onClick={handleDetails}>Details</Button>
+          {isEmployer && <Button onClick={handleApprove}>Approve</Button>}
```

Comment:
You can use util function here for checking and applying matched names

Reply:
Not sure if I totally understood. what names?

Reply:
Instead of duplication of code, you can do checking on a single Button component and add respectively its name and/or function call and etc.
for example:   <Button onClick={isEmployer ? handleApprove : other}>{isEmployer ? "Approve" : "other"}</Button>}

Reply:
or even use util function for whole logic, but this one can be consuming in terms of time. I think the above is easier.